target_redshift/__init__.py

```diff
@@ -340,17 +341,60 @@ def load_stream_batch(stream, records_to_load, row_count, db_sync, delete_rows=F
         row_count[stream] = 0
 
 
-def flush_records(stream, records_to_load, row_count, db_sync):
-    csv_fd, csv_file = mkstemp()
-    with open(csv_fd, 'w+b') as f:
-        for record in records_to_load.values():
-            csv_line = db_sync.record_to_csv_line(record)
-            f.write(bytes(csv_line + '\n', 'UTF-8'))
 
-    s3_key = db_sync.put_to_s3(csv_file, stream, row_count)
-    db_sync.load_csv(s3_key, row_count)
-    os.remove(csv_file)
-    db_sync.delete_from_s3(s3_key)
+def chunk_iterable(iterable, size):
+    """Yield successive n-sized chunks from iterable. The last chunk is not padded"""
+    iterable = iter(iterable)
+    return iter(lambda: tuple(islice(iterable, size)), ())
+
+
+def ceiling_division(n, d):
+    """Returns integer ceiling division of n / d"""
+    return -(n // -d)
+
+
+def flush_records(config, stream, records_to_load, row_count, db_sync):
```

Comment:
list of arguments changed here by adding `config` arg but not at the places where it's been called.

We need to pass the entire config map or preferably (??) only a new `slices` optional arg to the load_stream_batch function

Reply:
this looks very similar problem in #32. Can you please have a look? Otherwise it looks good.